services/src/debug.c

```diff
@@ -5,90 +5,42 @@
  *      Author: david_s5
  */
 
-#include <stdio.h>
-#include <stdint.h>
-#include <stdarg.h>
-#include <string.h>
-#include "config.h"
-#include "spark_macros.h"
 #include "debug.h"
-#include "timer_hal.h"
 
-LoggerOutputLevel log_level_at_run_time = LOG_LEVEL_AT_RUN_TIME;
-debug_output_fn debug_output_;
+#if defined(DEBUG_BUILD)
+#define LOG_LEVEL_AT_RUN_TIME LOG_LEVEL
+#endif
 
-void set_logger_output(debug_output_fn output, LoggerOutputLevel level)
-{
-    if (output)
-        debug_output_ = output;
-    if (level==DEFAULT_LEVEL)
-        level = LOG_LEVEL_AT_RUN_TIME;
-    log_level_at_run_time = level;
-}
+#if defined(RELEASE_BUILD)
+#define LOG_LEVEL_AT_RUN_TIME ERROR_LEVEL
+#endif
 
-void log_print_(int level, int line, const char *func, const char *file, const char *msg, ...)
-{
-    if (level<log_level_at_run_time || !debug_output_)
-        return;
+LoggerOutputLevel log_compat_level = LOG_LEVEL_AT_RUN_TIME;
+debug_output_fn log_compat_callback = NULL;
 
-    char _buffer[MAX_DEBUG_MESSAGE_LENGTH];
-    static char * levels[] = {
-            "",
-            "LOG  ",
-            "DEBUG",
-            "INFO ",
-            "WARN ",
-            "ERROR",
-            "PANIC",
-    };
-    va_list args;
-    va_start(args, msg);
-    file = file ? strrchr(file,'/') + 1 : "";
-    int trunc = snprintf(_buffer, arraySize(_buffer), "%010u:%s: %s %s(%d):", (unsigned)HAL_Timer_Get_Milli_Seconds(), levels[level/10], func, file, line);
-	debug_output_(_buffer);
-	if (trunc > arraySize(_buffer))
-	{
-		debug_output_("...");
-	}
-    trunc = vsnprintf(_buffer,arraySize(_buffer), msg, args);
-	debug_output_(_buffer);
-	if (trunc > arraySize(_buffer))
-	{
-		debug_output_("...");
-	}
-	debug_output_("\r\n");
-	va_end(args);
+void set_logger_output(debug_output_fn output, LoggerOutputLevel level) {
+    if (output) {
+        log_compat_callback = output;
+    }
+    if (level == DEFAULT_LEVEL) {
+        level = LOG_LEVEL_AT_RUN_TIME;
+    }
+    log_compat_level = level;
 }
 
-void log_print_direct_(int level, void* reserved, const char *msg, ...)
-{
-    if (level<log_level_at_run_time || !debug_output_)
-        return;
+void log_print_(int level, int line, const char *func, const char *file, const char *msg, ...) {
+    // Not implemented, use newer log_message() function instead
+}
 
-    char _buffer[MAX_DEBUG_MESSAGE_LENGTH];
-    va_list args;
-    va_start(args, msg);
-    int trunc = vsnprintf(_buffer, arraySize(_buffer), msg, args);
-	debug_output_(_buffer);
-	if (trunc > arraySize(_buffer))
-	{
-		debug_output_("...");
-	}
-	va_end(args);
+void log_print_direct_(int level, void* reserved, const char *msg, ...) {
+    // Not implemented, use newer log_printf() function instead
```

Comment:
// Deprecated, use newer log_printf() function instead
    // This function may not be removed because it is exported in dynalib